lib/generate_integration_url.py

```diff
@@ -0,0 +1,68 @@
+from string import Template
+from helpers import query
+from escape_helpers import sparql_escape_string, sparql_escape_uri
+from signinghub_api_client.client import SigningHubSession
+from . import exceptions, helpers, uri, validate, get_signflow_pieces
+
+# TODO: validation:
+# - piece is uploaded to signinghub
+def generate_integration_uri(signinghub_session: SigningHubSession,
+    signflow_uri: str, piece_uri: str,
+    collapse_panels: bool):
+    validate.ensure_signflow_exists(signflow_uri)
+    validate.ensure_piece_exists(piece_uri)
+
+    pieces = get_signflow_pieces.get_signflow_pieces(signflow_uri)
+    piece = helpers.ensure_1(pieces)
+    if piece["uri"] != piece_uri:
+        raise exceptions.InvalidArgumentException(f"Piece <{piece_uri}> is not linked to signflow <{signflow_uri}>.")
+
+    query_command = _query_signinghub_document.safe_substitute(
```

Comment:
Als ik de 2e paragraaf van [safe_substitute](https://docs.python.org/3.8/library/string.html#string.Template.safe_substitute) lees, had ik liefst toch gewoon `substitute` gebruikt, zoals [beschreven](https://github.com/MikiDi/mu-python-template/#writing-sparql-queries) in de template doc.

Reply:
Juist, had dit ook reeds gezien. Het staat in de "Offer for signature"-branch.